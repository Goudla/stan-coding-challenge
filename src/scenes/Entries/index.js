import React, { Component } from 'react';
import Helmet from 'react-helmet';

// Components
import Container from '../../components/Container';
import ErrorGate from '../../components/ErrorGate';
import InitialDataLoadedGate from '../../components/InitialDataLoadedGate';
import Message from '../../components/Message';
import PageContainer from '../../components/PageContainer';
import SubNavbar from '../../components/SubNavbar';
import SubNavbarHeading from '../../components/SubNavbarHeading';
import EntriesList from './components/EntriesList';
import EntriesRequest from './components/EntriesRequest';

type Props = {
  title: string,
  programType: string
};

class Entries extends Component<Props> {
  render() {
    const { title, programType } = this.props;
    return (
      <div>
        <Helmet>
          <title>{title}</title>
        </Helmet>
        <EntriesRequest />
        <SubNavbar>
          <Container>
            <SubNavbarHeading>Popular {title}</SubNavbarHeading>
          </Container>
        </SubNavbar>
        <PageContainer>
          <ErrorGate requestName="LOAD_PHOTOS">
            <InitialDataLoadedGate
              requestName="LOAD_PHOTOS"
              LoadingComponent={<Message>Loading...</Message>}>
              <EntriesList programType={programType} />
            </InitialDataLoadedGate>
          </ErrorGate>
        </PageContainer>
      </div>
    );
  }
}

export default Entries;
