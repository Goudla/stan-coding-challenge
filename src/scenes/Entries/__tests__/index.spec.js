import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

// Scenes
import Entries from '../';

describe('<Entries />', () => {
  it('Should render', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Entries
        title="Series"
        programType="series" />
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
