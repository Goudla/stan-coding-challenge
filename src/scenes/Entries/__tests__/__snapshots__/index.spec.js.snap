// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Entries /> Should render 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Series
    </title>
  </HelmetWrapper>
  <Connect(EntriesRequest) />
  <SubNavbar>
    <Container>
      <SubNavbarHeading>
        Popular 
        Series
      </SubNavbarHeading>
    </Container>
  </SubNavbar>
  <PageContainer>
    <Connect(ErrorGate)
      requestName="LOAD_PHOTOS"
    >
      <Connect(InitialDataLoadedGate)
        LoadingComponent={
          <Message>
            Loading...
          </Message>
        }
        requestName="LOAD_PHOTOS"
      >
        <Connect(EntriesList)
          programType="series"
        />
      </Connect(InitialDataLoadedGate)>
    </Connect(ErrorGate)>
  </PageContainer>
</div>
`;
